frontend/views/components/ButtonSubmit.vue

```diff
@@ -0,0 +1,50 @@
+<template lang='pug'>
+  button.is-loader(
+    v-bind='$attrs'
+    v-on='bindListeners'
+    type='submit'
+    :data-loading='ephemeral.isSubmitting'
+    :disabled='disabled || ephemeral.isSubmitting'
+  )
+    slot
+</template>
+
+<script>
+export default {
+  name: 'ButtonSubmit',
+  props: {
+    disabled: Boolean
+  },
+  data: () => ({
+    ephemeral: {
+      isSubmitting: false
+    }
+  }),
+  methods: {
+    async submit (e) {
+      if (this.ephemeral.isSubmitting) {
+        return
+      }
+
+      this.ephemeral.isSubmitting = true
+
+      // this.$listeners can await for async events.
+      // An advantage over $emit()
+      // More at: https://stackoverflow.com/questions/60554270/vuejs-difference-between-emit-and-listeners
```

Comment:
This stackoverflow link isn't needed, since I think you figured out the perfect solution all on your own! 😄 

Well done thinking to use `this.$listeners` like this, very clean, and yes, means we don't need to use `$refs`. 👍 

Can you instead include a link to the section in our style guide that talks about how to use this component?

Reply:
Done, please check it out. I also added an example on our Design System page. :)